### PR TITLE
Add new `POST /api/v1/datasets/{dataset_id}/records/search` endpoint with vector search support for admin (SDK) operations

### LINKED ISSUE #4114: [FEATURE] Add new `POST /api/v1/datasets/{dataset_id}/records/search` endpoint with vector search support for admin (SDK) operations
Refs https://github.com/argilla-io/argilla/issues/4020

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR add the following changes:
* Handler function `search_dataset_records` has been renamed to `search_current_user_dataset_records`.
* Add a new `POST /api/v1/datasets/{dataset_id}/records/search` endpoint without current user context so it can be used to search for records by owners and admin and getting responses from all users.
* Add a new `search_records_with_all_responses` method for dataset policies v1 used by the new endpoint.
* Update SDK API to use this new endpoint instead of the old one with current user context.
* Some small refactors:
  * Improve order on some policies functions.
  * Fix problem with `limit` query param not including a lower limit of 1.
  * Fix problem with `limit` query param using `lte` inexistent parameter (instead of the right one `le`).

I'm adding as well a new test type where we focus in testing one specific endpoint. This test includes a class with two important aspects:
* A instance method `url` defining the `url` that is used for all the tests.
* Simple and isolated tests that don't use external helper functions.

Closes #4114 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Running test suite locally.

**Checklist**

- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: src/argilla/client/sdk/v1/datasets/api.py [hunk 1]
```diff
@@ -234,7 +233,7 @@ def search_records(
     metadata_filters: Optional[List[str]] = None,
     limit: int = 50,
 ) -> Response[Union[FeedbackRecordsSearchModel, ErrorMessage, HTTPValidationError]]:
-    """Sends a POST request to `/api/me/datasets/{id}/records/search` endpoint to search for records inside an specific dataset.
+    """Sends a POST request to `/api/datasets/{id}/records/search` endpoint to search for records inside an specific dataset.
```
Comment (frascuchon) [inline src/argilla/client/sdk/v1/datasets/api.py]:
```suggestion
    """Sends a POST request to `/api/v1/datasets/{id}/records/search` endpoint to search for records inside an specific dataset.
```

File: src/argilla/server/apis/v1/handlers/datasets.py [hunk 2]
```diff
@@ -779,10 +779,11 @@ async def search_dataset_records(
     include: Optional[RecordIncludeParam] = Depends(parse_record_include_param),
     response_statuses: List[ResponseStatusFilter] = Query([], alias="response_status"),
     offset: int = Query(0, ge=0),
-    limit: int = Query(default=LIST_DATASET_RECORDS_LIMIT_DEFAULT, lte=LIST_DATASET_RECORDS_LIMIT_LTE),
+    limit: int = Query(default=LIST_DATASET_RECORDS_LIMIT_DEFAULT, ge=1, le=LIST_DATASET_RECORDS_LIMIT_LE),
```
Comment (frascuchon) [inline src/argilla/server/apis/v1/handlers/datasets.py:767]: We should add a breaking change tag using old clients (which send limit=0)

File: src/argilla/server/policies.py [hunk 3]
```diff
@@ -265,7 +265,7 @@ async def is_allowed(actor: User) -> bool:
         return is_allowed
 
     @classmethod
-    def create_vector_settings(cls, dataset: Dataset) -> PolicyAction:
+    def create_metadata_property(cls, dataset: Dataset):
```
Comment (gabrielmbmb) [inline src/argilla/server/policies.py]:
```suggestion
    def create_metadata_property(cls, dataset: Dataset) -> PolicyAction:
```

File: tests/unit/server/api/v1/datasets/test_search_dataset_records.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+from argilla.server.enums import RecordInclude
+from argilla.server.search_engine import SearchEngine, SearchResponseItem, SearchResponses
+from httpx import AsyncClient
+
+from tests.factories import AdminFactory, AnnotatorFactory, DatasetFactory, OwnerFactory, RecordFactory, ResponseFactory
+
+
+@pytest.mark.asyncio
+class TestSearchDatasetRecords:
+    def url(self, dataset_id: UUID):
```
Comment (gabrielmbmb) [inline tests/unit/server/api/v1/datasets/test_search_dataset_records.py]:
```suggestion
    def url(self, dataset_id: UUID) -> str:
```

**Review by gabrielmbmb — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -20,6 +20,7 @@ These are the section headers that we use:
 
 - Added `max_records` argument to `pull()` method for `RemoteFeedbackDataset`.([#4074](https://github.com/argilla-io/argilla/pull/4074))
 - Added functionality to push your models to huggingface hub with `ArgillaTrainer.push_to_huggingface` ([#3976](https://github.com/argilla-io/argilla/pull/3976)).
+- Added `POST /api/v1/datasets/:dataset_id/records/search` endpoint to search for records without user context, including responses by all users. ([#4143](https://github.com/argilla-io/argilla/pull/4143))
```
Comment (frascuchon) [inline CHANGELOG.md:30]: I think we should also add changes to the limit for existing endpoints.

**Review by frascuchon — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (jfcalvo) [reply CHANGELOG.md:30]: Yes, let me add it.

File: src/argilla/server/apis/v1/handlers/datasets.py (hunk 2 above)
Comment (jfcalvo) [reply src/argilla/server/apis/v1/handlers/datasets.py:767]: With this do you mean a new line on  `CHANGELOG.md`?

File: CHANGELOG.md (hunk 5 above)
Comment (jfcalvo) [reply CHANGELOG.md:30]: Done!

**Pull request merged**